# Домашнє завдання до теми "Python Basics"
В цьому ДЗ важливою умовою виконання завдань є те, що ми не імпортуємо жодних бібліотек, а робимо все на чистому Python, використовуючи лише базовий функціонал. Це необхідно аби відпрацювати роботу з базовими типами даних. А використовувати бібліотеки ми ще навчимось далі :)

## Робота з числами та арифметичними операторами

1.Створіть змінну, яка буде зберігати кількість яблук. І надайте їй значення 5. Виведіть значення змінної на екран.

In [1]:
apple = 5

In [2]:
apple

5

2.Виведіть тип змінної, створеної в завданні (1).

In [5]:
type(apple)

int

3.Додайте до значення змінної, створеної в завданні (1) число `12` і виведіть результат на екран з допомогою команди `print()`.

In [6]:
apple = apple + 12

In [7]:
print(apple)

17


4.Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [13]:
a = 101
b = 210
c = 111
d = 402

In [16]:
e = (a + b) * c / d

In [17]:
print(e)

85.8731343283582


Очікувана відповідь: 85.8731343283582

5.Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [18]:
a = 11283
b = 89

In [21]:
c = a % b

In [22]:
print(c)

69


6.Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [27]:
a = 11283
b = 89

In [28]:
c = a // b

In [29]:
print(c)

126


# Робота з рядками

7.Запишіть у змінну `name` ваше імʼя.

In [47]:
name = "Kateryna"

8.Запишіть у змінну `surname` ваше прізвище.

In [48]:
surname = "Fil"

9.Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [62]:
name_surname = name + " " + surname

In [63]:
name_surname

'Kateryna Fil'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva' 

10.Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [100]:
name_surname = 'Kateryna Fil'

In [102]:
print(s.upper())

KATERYNA FIL


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

11.Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [103]:
s[3]

'e'

Очікувана відповідь (тільки з вашими даними): 'n'

12.Виведіть першу та останню літеру вашого прізвища.

In [105]:
s[9]

'F'

In [106]:
s[11]

'l'

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

13.Виведіть довжину вашого прізвища.

In [109]:
s = "Fil"
len(s)

3

Очікувана відповідь (тільки з вашими даними): 7

14.Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [110]:
s = "Aaaa"
len(s)

4

Очікувана відповідь: 4

15.Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [124]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

In [127]:
my_string

'The quick brown fox jumps over the lazy dog!!!'

16.Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [7]:
print('The_quick_brown_fox_jumps_over_the_lazy_dog!!!')

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`

# Робота зі колекціями

У нас є наступний список:

In [73]:
my_list = [
    78, 12, 14, 27, 47, 28, 51, 46, 94, 37, 42, 20, 52, 33, 11, 11, 51,
    20, 81, 54, 86, 51, 38, 42, 49, 99, 76, 17,  9, 23, 97, 29, 29, 63,
    40, 11, 31, 76, 15, 21, 29, 75, 35,  2, 10, 71, 17, 10, 13,  7, 44,
    12, 19, 61, 13, 25, 62, 46, 10, 44, 75, 57, 20, 57,  3, 44, 16, 28,
    79, 57, 62, 70, 95,  8, 47, 82, 80, 45, 36, 44, 79, 95, 73, 42, 29,
    83, 80, 66, 56,  8, 73,  5, 99, 65, 65, 98, 17, 36, 16, 60
]

Будемо з ним працювати в наступних завданнях.

17.Виведіть 5 перших елементів списку `my_list` з допомогою слайсингу.

In [18]:
my_list[0:5]

[78, 12, 14, 27, 47]

Очікувана відповідь: [78, 12, 14, 27, 47]

18.Виведіть 3 останніх елементи списку `my_list` з допомогою слайсингу.

In [57]:
my_list[-3:]

[36, 16, 60]

19.Відсортуйте за зростанням список `my_list`, запишіть результат у змінну `my_list_sorted` та виведіть значення нової змінної на екран.

In [79]:
sorted(my_list)

[2,
 3,
 5,
 7,
 8,
 8,
 9,
 10,
 10,
 10,
 11,
 11,
 11,
 12,
 12,
 13,
 13,
 14,
 15,
 16,
 16,
 17,
 17,
 17,
 19,
 20,
 20,
 20,
 21,
 23,
 25,
 27,
 28,
 28,
 29,
 29,
 29,
 29,
 31,
 33,
 35,
 36,
 36,
 37,
 38,
 40,
 42,
 42,
 42,
 44,
 44,
 44,
 44,
 45,
 46,
 46,
 47,
 47,
 49,
 51,
 51,
 51,
 52,
 54,
 56,
 57,
 57,
 57,
 60,
 61,
 62,
 62,
 63,
 65,
 65,
 66,
 70,
 71,
 73,
 73,
 75,
 75,
 76,
 76,
 78,
 79,
 79,
 80,
 80,
 81,
 82,
 83,
 86,
 94,
 95,
 95,
 97,
 98,
 99,
 99]

In [81]:
my_list_sorted = sorted(my_list)

In [82]:
my_list_sorted

[2,
 3,
 5,
 7,
 8,
 8,
 9,
 10,
 10,
 10,
 11,
 11,
 11,
 12,
 12,
 13,
 13,
 14,
 15,
 16,
 16,
 17,
 17,
 17,
 19,
 20,
 20,
 20,
 21,
 23,
 25,
 27,
 28,
 28,
 29,
 29,
 29,
 29,
 31,
 33,
 35,
 36,
 36,
 37,
 38,
 40,
 42,
 42,
 42,
 44,
 44,
 44,
 44,
 45,
 46,
 46,
 47,
 47,
 49,
 51,
 51,
 51,
 52,
 54,
 56,
 57,
 57,
 57,
 60,
 61,
 62,
 62,
 63,
 65,
 65,
 66,
 70,
 71,
 73,
 73,
 75,
 75,
 76,
 76,
 78,
 79,
 79,
 80,
 80,
 81,
 82,
 83,
 86,
 94,
 95,
 95,
 97,
 98,
 99,
 99]

20.Виведіть максимальний і мінімальний елемент у списку `my_list`. Сформуйте відповідь у наступному форматі
```
Max element  <число>
Min element  <число>
```
Можна скористатись результатом з попереднього завдання, або спеціальними фукнціями.

In [97]:
print('Max element ', max(my_list))

print('Min element ', min(my_list))

Max element  99
Min element  2


Очікувана відповідь: 

```
Max element  99
Min element  2
```

In [99]:
21.Створіть новий список `my_list_2`, який містить кожен 11-й елемент списку `my_list`. Виведіть `my_list_2` на екран.  
Памʼятаємо, що перший з "кожного одинадцятого елементу" списку має індекс 10.
Для виконання завдання використайте слайсинг.

SyntaxError: invalid decimal literal (1413878862.py, line 1)

Очікуваний результат: `[42, 51, 29, 2, 13, 44, 80, 66, 16]`

22.Заданий список `new_list`. Перетворіть його на набір (set). Які зміни у наборі елементів Ви помітили? Відповідь напишіть у коментарі до коду (саме в клітинці з кодом). 

In [101]:
new_list = [22, 11, 67, 11, 88, 3213, 12, 78, 67, 11, 16, 15, 88]

In [102]:
new_list

[22, 11, 67, 11, 88, 3213, 12, 78, 67, 11, 16, 15, 88]

In [104]:
set(new_list)  # в set немає повторень і все стало впорядковано

{11, 12, 15, 16, 22, 67, 78, 88, 3213}

23.Аби отримати скільки разів зустрічається деякий елемент в списку, зручно використовувати метод count(). Працює це наступним чином:
```
some_list = [1,1,1,2,2]
some_list.count(1)
# 3
```

Виведіть, скільки разів зустрічається число 11 у списку `new_list`.

In [107]:
new_list.count(11)

3

24.Наостанок, попрактикуємось працювати зі словниками. Це завдання включає 4 невеличких пункти:

1. Створіть словник, що містить 5 пар ключ-значення, де ключі - це імена студентів (типу рядок), а значення - їхні оцінки з програмування від 1 до 12. Запишіть словник у змінну.
2. Виведіть весь словник на екран. Важливо, аби словник вивівся на екран в jupyter notebook, будь-який спосіб, який це робить, нам підходить.
3. Додайте до словника нового студента та його оцінку.
4. Виведіть оцінку одного конкретного студента, звертаючись до неї за його ім'ям.

In [110]:
new_dict = {'Mary':10, 'Nastya':11, 'Sofa':12, 'Olena':9, 'Marta':10}

In [111]:
new_dict

{'Mary': 10, 'Nastya': 11, 'Sofa': 12, 'Olena': 9, 'Marta': 10}

In [113]:
new_dict['lola'] = 11

In [114]:
new_dict

{'Mary': 10, 'Nastya': 11, 'Sofa': 12, 'Olena': 9, 'Marta': 10, 'lola': 11}

In [118]:
new_dict['Sofa']

12

# Ураа!! Ви завершили перше домашнє завдання!!!  
Аби здати ДЗ - залийте ноутбук у гітхаб репозиторій, який ви створили у перші дні навчання на курсі, та додайте посилання на файл у відповідному завданні на навчальній платформі.
Так тримати! 🚀🚀🚀